esmvaltool/diag_scripts/weighting/climwip.py

```diff
@@ -418,19 +439,53 @@ def visualize_and_save_weights(weights: 'xr.DataArray', cfg: dict,
     log_provenance(caption, filename_data, cfg, ancestors)
 
 
+def get_variable_groups(contribution: str, cfg: dict) -> list:
+    """Return a list of variable groups with contributions > 0."""
+    if cfg.get(contribution, None) is not None:
+        return [key for key, value in cfg[contribution].items() if value > 0]
+    return []
```

Comment:
This suggestion only makes sense in combination with some other suggestions. Basically my proposal is to use the contributions dicts as the key flag for deciding all your if statements.

```suggestion
def parse_contributions(metric: str, cfg: dict) -> list:
    """Only retain variable groups with performance or independence contributions > 0."""
    if cfg.get(f'{metric}_contributions') is not None:
        return {key: value for key, value in cfg[contribution].items() if value > 0}
    return {}
```